holoviews/core/util.py

```diff
@@ -29,20 +29,25 @@
 
 
 
-class HashGenerator(json.JSONEncoder):
+class HashableJSON(json.JSONEncoder):
     """
     Extends JSONEncoder to generate a hashable string for as many types
-    of key as possible. These keys are supplied by widgets and Stream
-    parameters and may be nested. If an unrecognized object type is
-    supplied, the objects id is used - this may be an issue when dealing
-    with mutable objects. For this reason, it is important to support
-    common mutable types such as pandas Dataframes and numpy arrays
-    (supported).
+    of object as possible including nested objects and objects that are
+    not normally hashable. The purpose of this class is to generate
+    unique strings that once hashed are suitable for memoization.
```

Comment:
Strictly speaking, it's not the hashed strings that are memoized, it's the actual results (and only implicitly even those, in this case). The hashed strings are *useful* for memoization, but they themselves are not memoized.

So maybe "The purpose of this class is to generate unique strings that once hashed are suitable for use in memoization and other cases where deep equality must be tested without storing the entire object".